estate_account/models/estate_property.py

```diff
@@ -0,0 +1,45 @@
+from odoo import Command, models
+from odoo.exceptions import AccessError
+from odoo.tools.translate import _
+
+
+class PropertyModel(models.Model):
+    _inherit = "estate.property"
+
+    def _fill_invoice(self, lines):
+        self.ensure_one()
+        return {
+            "move_type": "out_invoice",
+            "invoice_user_id": self.env.uid,
+            "partner_id": self.buyer_id.id,
+            "invoice_origin": self.name,
+            "invoice_line_ids": [Command.create(line) for line in lines],
+        }
+
+    def _create_invoice_line(self, name, quantity, unit_price):
+        return {
+            "name": name,
+            "quantity": quantity,
+            "price_unit": unit_price,
+        }
+
+    def _create_invoices(self):
+        if not self.env["account.move"].check_access_rights("create", False):
+            try:
+                self.check_access_rights("write")
+                self.check_access_rule("write")
+            except AccessError:
+                return self.env["account.move"]
+
+        invoice_vals_list = [
+            self._create_invoice_line(f"{self.name} (" + _("6%") + ")", 1, self.selling_price * .06),
+            self._create_invoice_line(f"{self.name} (" + _("10 000 before") + ")", 1, 10_1000),
+        ]
+
+        moves = self.env["account.move"].create(self._fill_invoice(invoice_vals_list))
+        return moves
```

Comment:
The access rights are implictly checked by `create()` and you do not use `sudo()` so it feels a little bit overkill to check them again manually here but I can understand that you don't want to throw an exception to the user if he does not have sufficient privileges.

The part that is harder for me to understand is that you check create **and** write access. Why ? If he does not have a create rights, I don't understand what difference it makes that the user can write on existing `account.move`.

Overall, the idea to play around with acces rights is really cool for this part of the tutorial but I'd make it more straightforward.
```suggestion
    def _create_invoice(self):
        if self.env["account.move"].check_access_rights("create", False):
            invoice_vals_list = [
                self._create_invoice_line(f"{self.name} (" + _("6%") + ")", 1, self.selling_price * .06),
                self._create_invoice_line(f"{self.name} (" + _("10 000 before") + ")", 1, 10_1000),
            ]
            return self.env["account.move"].create(self._fill_invoice(invoice_vals_list))
        return self.env["account.move"]
```